src/views/Dashboard.vue

```diff
@@ -0,0 +1,120 @@
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<DashboardWidget :items="importantMessages"
+		:show-more-url="''"
+		:loading="loading"
+		@hide="() => {}"
+		@markDone="() => {}">
+		<template v-slot:default="{ item }">
+			<DashboardWidgetItem :item="getWidgetItem(item)">
+				<template v-slot:avatar>
+					<Avatar v-if="item.from" :email="item.from[0].email" :display-name="item.from[0].label" />
+				</template>
+			</DashboardWidgetItem>
+		</template>
+		<template v-slot:empty-content>
+			<div class="empty-content">
+				<img class="empty-content__image" :src="emptyImage">
+				<p class="empty-content__text">
+					{{ t('mail', 'No messages found yet') }}
+				</p>
+				<a v-if="accounts.length === 2" :href="accountSetupUrl" class="button">{{ t('mail', 'Setup an account') }}</a>
+			</div>
+		</template>
+	</DashboardWidget>
+</template>
+
+<script>
+import { fetchEnvelopes } from '../service/MessageService'
+import { loadState } from '@nextcloud/initial-state'
+import { generateUrl, imagePath } from '@nextcloud/router'
+import Avatar from '../components/Avatar'
+import { DashboardWidget, DashboardWidgetItem } from '@nextcloud/vue-dashboard'
+
+const accounts = loadState('mail', 'mail-accounts')
+
+export default {
+	name: 'Dashboard',
+	components: {
+		Avatar,
+		DashboardWidget,
+		DashboardWidgetItem,
+	},
+	data() {
+		return {
+			messages: null,
+			accounts,
+			fetchedAccounts: 0,
+			emptyImage: imagePath('mail', 'newsletter.svg'),
+			accountSetupUrl: generateUrl('/apps/mail/#/setup'),
+		}
+	},
+	computed: {
+		loading() {
+			return this.fetchedAccounts < this.accounts.length
+		},
+		importantMessages() {
+			return this.messages ? this.messages.slice(0, 7) : []
+		},
+		getWidgetItem() {
+			return (item) => {
+				const { uid, accountId, mailbox } = item
+				return {
+					targetUrl: generateUrl(`/apps/mail/#/accounts/${accountId}/folders/${mailbox}/message/${accountId}-${mailbox}-${uid}`),
+					mainText: item.from ? item.from[0].label : '',
+					subText: item.subject,
+					message: item,
+				}
+			}
+		},
+	},
+	mounted() {
+		// TODO: check if there is a more sane way to query this and if other mailboxes should be fetched as well
+		this.accounts.forEach((account) => {
+			fetchEnvelopes(account.accountId, btoa('INBOX'), 'is:important', undefined, 10).then((messages) => {
+				messages = messages.map((message) => ({ ...message, accountId: account.accountId, mailbox: btoa('INBOX') }))
+				this.messages = this.messages !== null ? [...this.messages, ...messages] : messages
+				this.fetchedAccounts++
+			})
+		})
```

Comment:
Messages could be sorted by date. They are currently grouped by account.

They have a `dateInt` attribute.